games/CartPole-v0.jl

```diff
@@ -0,0 +1,94 @@
+import Reinforce
+using Reinforce:CartPoleV0, actions, reset!, finished, step!
+using Flux, StatsBase, Plots
+
+gr()
+
+#-------------Initialise game environment--------------#
+env = CartPoleV0()
+
+#---------------Parameters--------------#
+EPISODES = 3000
+STATE_SIZE = length(env.state)
+ACTION_SIZE = length(actions(env, env.state))
+
+MEM_SIZE = 2000
+BATCH_SIZE = 32
+
+γ = 0.95                # discount rate
+η = 0.001               # learning rate
+
+ϵ = 1.0                 # exploration rate
+ϵ_min = 0.01            # exploration minimum
+ϵ_decay = 0.995         # exploration decay
+
+memory = []             # used to remember past results
+
+#-------------Model Architecture--------------#
+model = Chain(Dense(STATE_SIZE, 24), Dense(24, 24), Dense(24, ACTION_SIZE))
+loss(x, y) = Flux.mse(model(x), y)
+opt = ADAM(params(model), η)
```

Comment:
The optimizer can be updated as per the current API, which does not take params as input.

Reply:
I don't think that would work as the definition of ADAM when checked in REPL turns out to be ```ADAM(params, η = 0.001; β1 = 0.9, β2 = 0.999, ϵ = 1e-08, decay = 0)``` where it is necessary to pass params.

Reply:
The new API is [here](https://github.com/FluxML/Flux.jl/blob/master/src/optimise/optimisers.jl).